Protocol/Error Messages/Protocol/QActions/QAction/CSharpCheckUnrecommendedFinalizer.cs

```diff
@@ -0,0 +1,56 @@
+// <auto-generated>This is auto-generated code by Validator Management Tool. Do not modify.</auto-generated>
+namespace Skyline.DataMiner.CICD.Validators.Protocol.Tests.Protocol.QActions.QAction.CSharpCheckUnrecommendedFinalizer
+{
+    using System;
+    using System.Collections.Generic;
+
+    using Skyline.DataMiner.CICD.Models.Protocol.Read;
+    using Skyline.DataMiner.CICD.Validators.Common.Interfaces;
+    using Skyline.DataMiner.CICD.Validators.Common.Model;
+    using Skyline.DataMiner.CICD.Validators.Protocol.Common;
+    using Skyline.DataMiner.CICD.Validators.Protocol.Interfaces;
+
+    internal static class Error
+    {
+        public static IValidationResult UnrecommendedFinalizer(IValidate test, IReadable referenceNode, IReadable positionNode, string finalizerName, string qactionId)
+        {
+            return new ValidationResult
+            {
+                Test = test,
+                CheckId = CheckId.CSharpCheckUnrecommendedFinalizer,
+                ErrorId = ErrorIds.UnrecommendedFinalizer,
+                FullId = "3.41.1",
+                Category = Category.QAction,
+                Severity = Severity.Critical,
+                Certainty = Certainty.Certain,
+                Source = Source.Validator,
+                FixImpact = FixImpact.NonBreaking,
+                GroupDescription = "",
+                Description = String.Format("Finalizer '{0}' is unrecommended. QAction ID '{1}'.", finalizerName, qactionId),
+                HowToFix = "",
+                ExampleCode = "",
+                Details = "Finalizers introduce unnecessary risk of process crashes because they execute code on the finalizer thread. The significant performance impact arises from the delayed cleanup until the finalizer finalizes the object. Finalizers can clean up unmanaged resources if the dispose method is not called, but it is preferred to use a SafeHandle to avoid the need for a finalizer. For resource management, it is recommended to use the IDisposable interface and the dispose pattern instead. More information can be found on the Microsoft docs (https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/finalizers).",
```

Comment:
```suggestion
                Details = "Finalizers need careful implementation as any exception thrown in a finalizer will result in a process crashes as this code is executes by the finalizer thread. The performance impact arises from the delayed cleanup until the finalizer finalizes the object. Finalizers can clean up unmanaged resources in case the dispose method was not called, but it is preferred to use a SafeHandle to avoid the need for a finalizer. For resource management, it is recommended to use the IDisposable interface and the dispose pattern instead. More information can be found on the Microsoft docs (https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/finalizers).",
```

Reply:
Done in separate commit to not have a failing build.